inc/export/class-endpoint.php

```diff
@@ -0,0 +1,323 @@
+<?php
+/**
+ * REST API endpoint to retrieve all the Altis Analytics data for a particular date.
+ *
+ * Displays 5000 items at once, after that there's pagination.
+ */
+
+declare( strict_types=1 );
+
+namespace Altis\Analytics\Export;
+
+use Altis\Analytics\Utils;
+use DateTime;
+use Exception;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Server;
+
+use function HM\GTM\flatten_array as GTMFlatten_array;
+
+/**
+ * Class Data_Endpoint
+ *
+ * @package RN\Altis_Analytics_Customisations
+ */
+class Endpoint {
+	/**
+	 * Add hooks.
+	 */
+	public function bootstrap() : void {
+		add_action( 'rest_api_init', [ $this, 'register_routes' ] );
+		add_filter( 'rest_pre_serve_request', [ $this, 'pre_serve_request' ], 10, 4 );
+	}
+
+	/**
+	 * Registers the route.
+	 */
+	public function register_routes() : void {
+		register_rest_route(
+			'analytics/v1',
+			'export/(?P<date>\d{4}-\d{2}-\d{2})',
+			[
+				'args'   => [
+					'date' => [
+						'description' => 'Day to retrieve the analytics data from.',
+						'type' => 'date',
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+					'page' => [
+						'description'       => 'Current page',
+						'type'              => 'integer',
+						'default'           => 1,
+						'sanitize_callback' => 'absint',
+						'validate_callback' => 'rest_validate_request_arg',
```

Comment:
these are not needed when passing correct `type` -- I've seen this pattern some a few people though, do you know where it comes from?

Reply:
This was from the original code, not sure on the pattern itself. Maybe @fklein-lu would have more information